src/data_provider/src/processes/processInterfaceSolaris.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015, Wazuh Inc.
+ * March 9, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _PROCESS_INTERFACE_SOLARIS_H
+#define _PROCESS_INTERFACE_SOLARIS_H
+
+#include "iprocessInterface.h"
+#include "iprocessWrapper.h"
+
+class FactorySolarisProcess
```

Comment:
What is the use of this class?
Why don't directly create "SolarisProcessImpl" object in factory class "FactoryProcessFamilyCreator<OSType::SOLARIS>"?

Reply:
As a way to specialize the creation of wrapper

Reply:
Yes, I understand, but it could be easier

Reply:
This code is based on something already done. You can see more about this in the [network implementation](https://github.com/wazuh/wazuh/blob/16009-dataprovider-add-solaris-processes/src/data_provider/src/network/networkFamilyDataAFactory.h).